history.go

```diff
@@ -0,0 +1,75 @@
+package merkletree
+
+import (
+	"errors"
+
+	"github.com/coniks-sys/libmerkleprefixtree-go/crypto"
+)
+
+var (
+	ErrHistoryExisted = errors.New("[merkletree] trying to override existing history")
+	ErrBadEpoch       = errors.New("[merkletree] next epoch's STR has bad epoch ")
+)
+
+type History struct {
+	tree          *MerkleTree
+	currentSTR    *SignedTreeRoot
+	epochInterval int64
+}
+
+func NewHistory(m *MerkleTree, key crypto.SigningKey, startEp, epInterval int64) *History {
+	h := new(History)
+	h.tree = m
+	h.epochInterval = epInterval
+	h.currentSTR = newSTR(m, startEp, 0, make([]byte, crypto.HashSizeByte), key)
+	return h
+}
+
+func (h *History) UpdateHistory(m *MerkleTree, nextEp int64) error {
```

Comment:
> I was asking if that's something we should be doing for them though.

Yes. I mean there are some extra works need to be done by the developer before/after updating the tree. Of course we can setup the timer as a part of the lib but I think it would have no advantages here. I need to think more about it.